python/cuml/solvers/qn.pyx

```diff
@@ -206,8 +206,14 @@ class QN(Base,
         will not be regularized.
     max_iter: int (default = 1000)
         Maximum number of iterations taken for the solvers to converge.
-    tol: float (default = 1e-3)
-        The training process will stop if current_loss > previous_loss - tol
+    tol: float (default = 1e-4)
```

Comment:
- Comparison to Sklearn tolerance should be described in [logistic_regression.pyx](https://github.com/rapidsai/cuml/blob/562ea66841d7a41b4dd5e4b5ac7e3fbac69caa9f/python/cuml/linear_model/logistic_regression.pyx#L121). 
- Please compare here to gtol from scpipy.optimize.minimiz(method=’L-BFGS-B’).
- Consider presenting the actual formula: (norm(current_loss_grad, inf) <= tol * max(current_loss, tol)
- Apart from describing the difference to sklearn, it would be nice to add an action oriented item like : "To ensure that the convergence tolerance is tighter than Sklearn, set tol = tol_sklearn / n_samples / fmag" The problem is that we do not know fmag in advance.
  - We know that tol <= fmag <= O(1), this way we could argue that tol = tol_sklearn / n_samples would always leads to tighter tolerance than sklearn.
  -  Alternatively you could drop fmag, and compare norm(current_loss_grad, inf) <= tol. That way those who care about exact alignment of tolerance parameter, can easily achieve that. Would the default tol=1e-4 value be too large in that case to achieve comparable accuracy to sklearn? If it is too large, then it is probably better to keep relative tolerance.

Reply:
To summarize this&offline conversations:

  * I added only a mere reference in `logistic_regression.pyx` to `qn.pyx` for details about `tol` (also took the liberty fix couple markup typos there).
  * Elaborated the details and the difference to sklearn in `qn.pyx`
  * Suggested to divide the `tol` by the sample size.